### DISCUSSION TITLE
Fetching release notes from github app

### DISCUSSION THREAD

**Discussion by dbasilio** [renovatebot/renovate#10545] [Request Help]
We use the github-hosted Renovate app. We maintain some internal monorepos which publish their changelogs on the Github releases. I read through the section about [looking up release notes](https://docs.renovatebot.com/private-modules/#looking-up-release-notes) in the docs, but it's fairly specific to self-hosted renovate, not the github app. 

Is there any way to fetch the release notes from our github releases pages using the github app? Currently PRs from renovate for updates from our internal monorepos are not containing the release notes, and the logs for the job don't really specify why they don't.

**Comment by rarkins** [comment]
Assuming you mean "GitHub Releases" and not that you publish Eg CHANGELOGS.md files.

So you develop the packages all from the same private monotype but then you have additional public repos for release notes, one per package? If not then please describe in more detail.

**Reply by noahnu** [reply]
We work together so I can answer. Here's the setup:

- Private monorepo which publishes packages to a private registry.
  - Each package "published" from the monorepo gets an associated private GitHub Release in that same monorepo.
- Private repos (non-mononrepos) which pull in those private packages.

The monorepo doesn't have a dedicated changelog file, as we use GitHub releases to record changelogs, however we can add dedicated changelog files per package (`<repoRoot>/packages/<some-package>/CHANGELOG.md`) if something like that would work.

Here's an example of what a "release" looks like (from our open source project). Only difference is that the one we're referring to is private. https://github.com/tophat/monodeploy/releases/tag/%40monodeploy%2Fpublish%400.4.8

**Reply by rarkins** [reply]
So you use github releases in the monorepo itself, and each release is prefixed by the package name?

That *should* be supported through this rule: https://github.com/renovatebot/renovate/blob/aa41ae15bac6e685f0e3c805466ae0cbd841c536/lib/workers/pr/changelog/release-notes.ts#L113

**Reply by dbasilio** [reply]
Yeah it seems like it _should_ work but we're not getting release notes for any of our internal packages. Is there a way to view trace logs from the renovate dashboard, that might give us more insight into what's happening.

**Reply by rarkins** [reply]
We don't trace log in the app as it's very heavy, but maybe we can add a debug logs somewhere useful that won't be a burden.

**Reply by dbasilio** [reply]
@rarkins Any next steps we can take to try to hunt down the problem?

**Reply by rarkins** [reply]
You could follow our [local development](https://github.com/renovatebot/renovate/blob/main/docs/development/local-development.md) instructions so that you can `yarn debug --dry-run=true <downstream repo>` and then step through the release notes extraction logic mentioned above. Make sure that the token has access to both the downstream and upstream repos, plus the package registry too of course.

**Reply by rarkins** [reply]
Alternatively if that public repo you listed above is *exactly* how you name your monorepo releases then you can probably set up a simple public reproduction repo using that?

**Reply by dbasilio** [reply]
@rarkins I created a minimal repro on [this repo](https://github.com/dbasilio/renovate-bug-repro) including links and screenshots in the readme. Let me know if you need anything else!

**Reply by noahnu** [reply]
I stepped through it locally and it seems `sourceUrl` is always undefined so it exits early: https://github.com/renovatebot/renovate/blob/74d7691befa635c1220d8569936273bfb950e748/lib/workers/pr/changelog/index.ts#L17

**Reply by dbasilio** [reply]
Specifying a sourceUrl fails the validator though so we can't add it (we run the validator on our shared config repo)

**Reply by dbasilio** [reply]
@rarkins Anything we can do to help unblock this?

**Reply by rarkins** [reply]
Somebody needs to identify why sourceUrl is undefined. Maybe it's never returned at all by the registry? If not then the root cause is that the package doesn't declare its source repository in a way Renovate understands.

**Reply by noahnu** [reply]
Found the issue. We had the repository url set to `git@github.com:tophat/monodeploy.git` which renovate can't parse. Looks like it silently strips sourceUrls it doesn't understand. Would be great if it produced a warning.

Either way, this is something we'll change on our end. We should be using the repository object structure anyway to specify the repository subdirectory.

**Reply by rarkins** [reply]
Please file a bug issue with this URL format so that we can support it anyway. No reproduction necessary as we can verify it with unit tests only.

**Reply by noahnu** [reply]
https://github.com/renovatebot/renovate/issues/10717
